service/health/health.go

```diff
@@ -31,7 +35,36 @@ func NewRegistration() *serviceregistry.Service[grpchealth.Checker] {
 				if err != nil {
 					srp.Logger.Error("failed to set well-known config", slog.String("error", err.Error()))
 				}
-				return HealthService{logger: srp.Logger}, nil
+				hs := HealthService{logger: srp.Logger}
+				return hs, func(_ context.Context, mux *runtime.ServeMux) error {
+					err := mux.HandlePath(http.MethodGet, "/healthz", func(w http.ResponseWriter, r *http.Request, _ map[string]string) { //nolint:contextcheck // check is not relevant here
+						resp, err := hs.Check(context.Background(), &grpchealth.CheckRequest{
+							Service: r.URL.Query().Get("service"),
+						})
+						if err != nil {
+							http.Error(w, err.Error(), http.StatusInternalServerError)
+							return
+						}
+						status := map[string]interface{}{"status": strings.ToUpper(resp.Status.String())}
```

Comment:
Should we include the service name here if it is set?

Reply:
The `grpchealth.CheckResponse` only has `Status` built in.